go.mod

```diff
@@ -9,6 +9,7 @@ require (
 	github.com/go-openapi/spec v0.19.5
 	github.com/gogo/protobuf v1.3.1 // indirect
 	github.com/golang/groupcache v0.0.0-20191027212112-611e8accdfc9 // indirect
+	github.com/google/btree v1.0.0 // indirect
```

Comment:
I guess this is for my learning as much as anything else, but - how do we know this is necessary, and in general, do we have a way to verify all our dependencies are still necessary? I removed this line locally and was still able to successfully run `make build && make test validate && make ci && make e2e status=keep`. Thanks!

Reply:
`go build` adds missing dependencies if necessary, so removing indirect dependencies from `go.mod` doesn’t affect the build. Indirect dependencies are transitive dependencies pulled in by dependencies listed in `go.mod` which don’t yet use modules themselves; they are added to our `go.mod` to ensure that it provides a full record of our module dependencies. See [this entry in the modules wiki](https://github.com/golang/go/wiki/Modules#why-does-go-mod-tidy-record-indirect-and-test-dependencies-in-my-gomod) for details.

To address your main question, `go mod tidy` is the reference tool for maintaining `go.mod`. It will take care of removing obsolete dependencies.